packages/cli/src/commands/locations/rooms.ts

```diff
@@ -33,6 +33,26 @@ export type RoomWithLocation = Room & {
 	locationName?: string
 }
 
+//
+export async function chooseRoom(command: APICommand, locationId?: string, deviceFromArg?: string): Promise<[string, string]> {
+	const rooms = await getRoomsByLocation(command.client, locationId)
+	const config = {
+		itemName: 'room',
+		primaryKeyName: 'roomId',
+		sortKeyName: 'name',
+		listTableFieldDefinitions: tableFieldDefinitions,
+	}
+	const roomId = await selectFromList(command, config, deviceFromArg, async () => rooms)
+	const room = rooms.find(room => room.roomId === roomId)
+	if (!room) {
+		throw new CLIError(`could not find room with id ${roomId}`)
+	}
+	if (!room.locationId) {
+		throw new CLIError(`could not determine location id for room ${roomId}`)
+	}
+	return [roomId, room.locationId]
```

Comment:
I think in this case, it might be worth straying from the pattern of returning raw ids to just returning the room, since it has the room and location ids on it already and would closer match the name of the function.

Reply:
I actually started with that but for some reason `roomId` in `Room` is optional and since I know `roomId` here is defined, doing it this way saves the user a bit of type-checking.